cli/src/test/java/com/devonfw/tools/ide/context/IdeTestContext.java

```diff
@@ -87,4 +102,85 @@ public IdeTestLogger getLogger() {
 
     return logger;
   }
+
+  /**
+   * Reads the content of the given file and replaces the placeholder "${testbaseurl}" with the actual base URL. Copy from AbstractUrlUpdaterTest.
+   *
+   * @param file the {@link Path} to the file to read.
+   * @param wmRuntimeInfo the {@link WireMockRuntimeInfo} providing the base URL.
+   * @return the resolved file content.
+   */
+  public static String readAndResolve(Path file, WireMockRuntimeInfo wmRuntimeInfo) {
+
+    try {
+      String payload = Files.readString(file);
+      return payload.replace(ToolRepositoryMock.VARIABLE_TESTBASEURL, wmRuntimeInfo.getHttpBaseUrl());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void mockNpmPackageResponses(WireMockRuntimeInfo wireMockRuntimeInfo) {
+    Path npmRoot = this.getIdeHome()
+        .getParent()
+        .resolve("repository")
+        .resolve("npmjs");
+
+    try (Stream<Path> files = Files.walk(npmRoot)) {
+      files.filter(p -> Files.isRegularFile(p) && p.getFileName().toString().endsWith(".json"))
+          .forEach(jsonFile -> {
+            // Derive package path from relative file path, e.g.
+            //   <root>/@angular/cli.json  -> "/@angular/cli"
+            //   <root>/yarn.json          -> "/yarn"
+            Path rel = npmRoot.relativize(jsonFile);
+            String packagePath = "/" + rel.toString()
+                .replace(File.separatorChar, '/')
+                .replaceAll("\\.json$", "");
+
+            String body = readAndResolve(jsonFile, wireMockRuntimeInfo);
+
+            stubFor(get(urlPathEqualTo(packagePath))
+                .willReturn(aResponse()
+                    .withStatus(200)
+                    .withHeader("Content-Type", "application/json")
+                    .withBody(body)));
+
+          });
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
Can't we move this to `NpmRepositoryMock`?
Seems misplaced here...

Reply:
I've moved this to NpmRepositoryMock now.